src/lib/es2020.promise.d.ts

```diff
@@ -17,14 +17,13 @@ interface PromiseConstructor {
      * @param values An array of Promises.
      * @returns A new Promise.
      */
-    allSettled<T extends readonly unknown[] | readonly [unknown]>(values: T):
-        Promise<{ -readonly [P in keyof T]: PromiseSettledResult<T[P] extends PromiseLike<infer U> ? U : T[P]> }>;
+    allSettled<T extends readonly unknown[] | []>(values: T): Promise<{ -readonly [P in keyof T]: PromiseSettledResult<awaited T[P]> }>;

```

Comment:
The `readonly []` is there to preserve support for read‑only tuples:

```suggestion
    allSettled<T extends readonly unknown[] | readonly []>(values: T): Promise<{ -readonly [P in keyof T]: PromiseSettledResult<awaited T[P]> }>;
```

Reply:
I had a conversation about this with @DanielRosenwasser yesterday. Read-only tuples would still get caught by `readonly unknown[]`. The `| []` actually just forces contextual typing to treat `T` as a tuple, so the `readonly` keyword is unnecessary.